crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1285 @@
+    params: SmallVec<[Param; 3]>,
+    internal_name: String,
+}
+
+impl<'py> FromPyObject<'py> for NormalOperation {
+    fn extract(ob: &'py PyAny) -> PyResult<Self> {
+        let operation = convert_py_to_operation_type(ob.py(), ob.into())?;
+        let internal_name = ob
+            .getattr("base_class")?
+            .getattr("__name__")?
+            .extract::<String>()?;
+        Ok(Self {
+            operation: operation.operation,
+            params: operation.params,
+            internal_name,
+        })
+    }
+}
+
+impl IntoPy<PyObject> for NormalOperation {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        self.to_object(py)
+    }
+}
+
+impl ToPyObject for NormalOperation {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        operation_type_and_data_to_py(py, &self.operation, &self.params, None, None, None, None)
+            .ok()
+            .to_object(py)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct VariableOperation {
+    internal_name: String,
+    object: PyObject,
+}
+
+impl<'py> FromPyObject<'py> for VariableOperation {
+    fn extract(ob: &'py PyAny) -> PyResult<Self> {
+        Ok(Self {
+            internal_name: ob.getattr("__name__")?.extract::<String>()?,
+            object: ob.into(),
+        })
+    }
+}
+
+impl IntoPy<PyObject> for VariableOperation {
+    fn into_py(self, _py: Python<'_>) -> PyObject {
+        self.object
+    }
+}
+
+impl ToPyObject for VariableOperation {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        self.object.to_object(py)
+    }
+}
+
+/**
+A rust representation of a ``Target`` object.
+
+The intent of the ``Target`` object is to inform Qiskit's compiler about
+the constraints of a particular backend so the compiler can compile an
+input circuit to something that works and is optimized for a device. It
+currently contains a description of instructions on a backend and their
+properties as well as some timing information. However, this exact
+interface may evolve over time as the needs of the compiler change. These
+changes will be done in a backwards compatible and controlled manner when
+they are made (either through versioning, subclassing, or mixins) to add
+on to the set of information exposed by a target.
+ */
+#[pyclass(
+    mapping,
+    subclass,
+    name = "BaseTarget",
+    module = "qiskit._accelerate.target"
+)]
+#[derive(Clone, Debug)]
+pub struct Target {
+    #[pyo3(get, set)]
+    pub description: Option<String>,
+    #[pyo3(get)]
+    pub num_qubits: Option<usize>,
+    #[pyo3(get, set)]
+    pub dt: Option<f64>,
+    #[pyo3(get, set)]
+    pub granularity: i32,
+    #[pyo3(get, set)]
+    pub min_length: usize,
+    #[pyo3(get, set)]
+    pub pulse_alignment: i32,
+    #[pyo3(get, set)]
+    pub acquire_alignment: i32,
+    #[pyo3(get, set)]
+    // TODO: Port to Rust.
+    pub qubit_properties: Option<Vec<PyObject>>,
+    #[pyo3(get, set)]
+    pub concurrent_measurements: Vec<Vec<usize>>,
```

Comment:
This might make more sense, since it's a grouping of qubit indices.
```suggestion
    pub concurrent_measurements: Vec<Vec<PhysicalQubit>>,
```